server/services/DeckService.js

```diff
@@ -580,8 +580,8 @@ class DeckService {
         }
     }
 
-    solveEnhancements(list, deckCards) {
-        const cards = list.map((c) => deckCards.find((x) => x.id === c));
+    countEnhancements(list, deckCards) {
+        const cards = list.map((c) => deckCards.find((x) => x.id === c)).filter(Boolean);
```

Comment:
what is the meaning of filter(Boolean)?

Reply:
The list is the master vaults card array that also contains the tide card.  At this point the tide card has already been removed from deckCards and yields an 'undefined' in the find() function.

.filter(Boolean) removes that undefined.

Reply:
I do not understand why you pass list and deckCards to the same function. Why not work only with deckCards, which has tide removed right before you call countEnhancements.

Reply:
List is the array of 36 or 37 card uuids. deckCards is the card objects but only one of each, so it misses duplicated cards in the deck.  

I might overhaul how this all is processed after I get it working again

Reply:
it misses duplicates, but it guarantees enhanced cards are there, so you do not have to worry about the whole list.